src/DataManagement/Goals/GoalManager.cs

```diff
@@ -19,59 +22,116 @@ public GoalManager(IGoalList goalList, int userId = 0)
+        }
+
+        public bool Edit(IGoal oldGoal, IGoal updatedGoal)
+        {
+            return Edit(oldGoal.Id, updatedGoal);
         }
 
-        public bool Edit(IGoal oldGoal, IGoal newGoal)
+        public bool Edit(int id, IGoal updatedGoal)
         {
-            if (!GoalDbUpdater.Edit(oldGoal, newGoal))
+            if (!GoalDatabase.Update(id, UserId, updatedGoal))
             {
                 return false;
             }
-            var localGoal = GoalList.FirstOrDefault(x => x.Id == oldGoal.Id);
+            var localGoal = GoalList.FirstOrDefault(x => x.Id == id);
             if (localGoal is null)
             {
-                ExceptionHandler.Log("Edited goal id: " + oldGoal.Id + " in database but couldn't find it locally");
+                ExceptionHandler.Log("Edited goal id: " + id + " in database but couldn't find it locally");
                 return false;
             }
-            localGoal.Edit(newGoal);
+            localGoal.Edit(updatedGoal);
             return true;
         }
 
         public bool Remove(IGoal goal)
         {
-            if (!GoalDbUpdater.Remove(goal))
+            return Remove(goal.Id);
+        }
+
+        public bool Remove(int id)
+        {
+            if (!GoalDatabase.Delete(id, UserId))
             {
                 return false;
             }
-            var localGoal = GoalList.FirstOrDefault(x => x.Id == goal.Id);
+            var localGoal = GoalList.FirstOrDefault(x => x.Id == id);
             if (localGoal is null)
             {
-                ExceptionHandler.Log("Removed goal id: " + goal.Id  + " from database but couldn't find it locally" );
+                ExceptionHandler.Log("Removed goal id: " + id + " from database but couldn't find it locally");
                 return false;
             }
 
             GoalList.Remove(localGoal);
             return true;
         }
 
-        public bool ReadFromDb()
+        public bool RemoveAll(IGoalList entryList)
         {
-            using var db = new DatabaseContext();
+            var idList = entryList.Select(va => va.Id).ToArray();
 
-            foreach (var dbGoal in db.Goals.Where(x => x.UserId == UserId)) // query executed and data obtained from database
+            if (!GoalDatabase.DeleteList(idList, UserId))
             {
-                var newEntry = new Goal(dbGoal);
-                GoalList.Add(newEntry);
+                return false;
             }
 
+            var temp = new GoalList();
+            GoalList.RemoveAll(entryList.Contains);
             return true;
         }
+
+        public bool RemoveAll(IEnumerable<int> idList)
+        {
+            var idArray = idList as int[] ?? idList.ToArray();
+
+            if (!GoalDatabase.DeleteList(idArray, UserId))
+            {
+                return false;
+            }
+
+            var temp = new GoalList();
+            temp.AddRange(from entry in GoalList from id in idArray where entry.Id == id select entry);
```

Comment:
why is this used as querie and not simple linq?

Reply:
So this code stands for :
            foreach (var goal in GoalList)
            {
                foreach (var id in idArray)
                {
                    if (goal.Id == id)
                    {
                        temp.Add(goal);
                    }
                }
            }
It's kind of like joining two tables together, so I don't know how you'd have to write it as a usual lambda expression we use (this is code is reshaper by the way :D ). If you know how to write this in a simpler way I'm open to suggestions, as I'm really no expert in this subject

Reply:
Something like LINQ join which was covered in yesterday's lecture may be the solution?

Reply:
Actually it turns out this is (as I understand) the is the LINQ stuff that was covered in yesterday's lecture
Our code:

            temp.AddRange(
                from goal in GoalList 
                from id in idArray 
                where goal.Id == id 
                select goal);

Example given on LINQ Join:

            var personCountryInner = from p in persons
                join c in regions
                    on p.RegionId equals c.ID
                select new
                {
                    PersonName = p.Name,
                    RegionName = c.Name
                };